Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/mechanisms/ArmTeleOp.java

```diff
@@ -0,0 +1,249 @@
+    private double armGoalLocation;
+
+    //Used to make sure button inputs are only registered once
+    private boolean dpadDownPressed = false;
+
+    //Keeps track of if the arm is in stick control mode, or if it is in move to position mode.
+    private boolean dpadUpPressed = false;
+
+    //Keeps track of if the arm is in stick control mode, or if it is in move to position mode.
+    public static boolean usingStick = true;
+
+    //The PID constants for the arm
+    public static double armKp = 0.01, armKi = 0, armKd = 0;
+    public static double armAcceleration = 0.001, maxArmVelocity = 100; //The acceleration of the arm every second, and the maxim highest velocity the arm will accelerate to.
+    public static double armVelocity, currentArmLocation; //The velocity the code thinks the arm is at.
+    public enum armPhases { //The different phases the arm motion profiling can be in.
+        accelerating,
+        maintainingSpeed,
+        decelerating,
+        stopped
+    }
+    public static armPhases armPhase; //Keeps track of the mode the arm motion profiling is in.
+    public static double initArmLocation, armAccelerationOverDist; //The arm location at the start of the motion profile. The distance it took for the arm to accelerate.
+    private double lastArmProfileTime ;
+    private double lastPidTime, lastPidError, cumulativeError; //The time that the PID loop was last run. The error the last time the PID loop was run. The total error of the pid loop every time it was run.
+
+    public ArmTeleOp(Gamepad gamepad2, DcMotor armMotor, Servo dumperServo) {
+        //passes necessary API objects in the class.
+        this.gamepad2 = gamepad2;
+        this.armMotor = armMotor;
+        this.dumperServo = dumperServo;
+
+        //Init variables so they don't hold their value from last time the code was run
+        armGoalLocation = 0;
+        armVelocity = 0;
+        currentArmLocation = 0;
+        armPhase = armPhases.accelerating;
+        initArmLocation = this.armMotor.getCurrentPosition();
+        lastArmProfileTime = 0;
+        cumulativeError = 0;
+    }
+
+    private double armPID(double goalVelocity, double currentVelocity, double Kp, double Ki, double Kd) {
+
+        double currentTime, elapsedTime;
+        double error, rateError;
+        double output;
+
+        currentTime = time.seconds();
+        elapsedTime = currentTime - lastPidTime;
+
+        error = goalVelocity - currentVelocity;
+
+        if (Ki != 0)
+            cumulativeError += error * elapsedTime;
+
+        rateError = (error - lastPidError) / elapsedTime;
+
+        output = Kp * error + Ki * cumulativeError + Kd * rateError;
+
+        lastPidError = error;
+        lastPidTime = currentTime;
+
+        return output;
+    }
+
+    private void armPidMotionProfile(double goalLocation) {
+        double currentTime, elapsedTime; //currentTime: The current time sense boot, elapsedTime: the time sense the last loop of the program.
```

Comment:
It's really hard to read when in line comments are actually in line. Probably better to just put them on top if they're going to have this much explanation and not just be a couple words